src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,120 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[CAPACITY];
+
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size == table.length * LOAD_FACTOR) {
+            resize();
+        }
+        int index = getPositionInTable(key);
+        Node<K, V> current = table[index];
+
+        if (current == null) {
+            table[index] = new Node<>(key, value, null);
+        }
 
+        while (current != null) {
+
+            if (Objects.equals(current.getKey(), key)) {
+                current.setValue(value);
+                return;
+            }
+
+            if (current.getNext() == null) {
+                Node<K, V> newNode = new Node<>(key, value, null);
+                current.setNext(newNode);
+                break;
+            }
+            current = current.getNext();
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int index = getPositionInTable(key);
+        Node<K, V> node = table[index];
+        while (node != null) {
+            if (Objects.equals(key, node.getKey())) {
+                return node.getValue();
+            }
+            node = node.getNext();
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getPositionInTable(K key) {
+        if (key != null) {
+            int position = key.hashCode() % table.length;
+            if (position < 0) {
+                return position * (-1);
+            }
+            return position;
+        } else {
+            return 0;
+        }
```

Comment:
Let's rafactor this method and use just ternary operator and Math method, so that it would be just one line
```suggestion
        return key == null ? 0 : Math.abs(key.hashCode() % table.length);
```